### PR TITLE
Simplify handle locking

### PR DESCRIPTION
**Pull request by michaelcahill**
This is a step in the direction of removing the schema and table locks.  It eliminates a lot of locking code that has traditionally been difficult to modify.

refs WT-1598

### CODE REVIEW

File: src/session/session_dhandle.c [hunk 1]
```diff
@@ -131,22 +174,21 @@ __wt_session_lock_dhandle(WT_SESSION_IMPL *session, uint32_t flags, int *deadp)
...
 
-	locked = F_ISSET(dhandle, WT_DHANDLE_EXCLUSIVE) ? WRITELOCK : READLOCK;
 	/*
 	 * If we had special flags set, close the handle so that future access
 	 * can get a handle without special flags.
 	 */
 	if (F_ISSET(dhandle, WT_DHANDLE_DISCARD_FORCE)) {
-		WT_WITH_DHANDLE_LOCK(session,
-		    ret = __wt_conn_btree_sync_and_close(session, 0, 1));
+		ret = __wt_conn_btree_sync_and_close(session, 0, 1);
```
Comment (michaelcahill) [inline src/session/session_dhandle.c]: This is the fix for WT-1910.

**Comment by keithbostic** [comment]
autobuild: retest this please

**Comment by michaelcahill** [comment]
Note that this is not yet ready to merge: there is a deadlock relating to opening bulk cursors...

**Comment by michaelcahill** [comment]
@keithbostic, I think this is ready to go -- I took the opportunity to do some renaming, which is why this looks big, but actually is gets rid of a lot of code from `conn_dhandle.c`.

Unfortunately, getting it stable required putting back in the locking during handle open.  I'm going to take another pass at getting rid of that under the banner of WT-1598 when I get a chance, but this branch should get tests running against develop again.

Can you please review and merge if you're happy?

**Comment by keithbostic** [comment]
@michaelcahill, this is a nice change, I like it a lot (the only thing that makes me nervous is calling `__wt_conn_btree_open` from the session layer, that doesn't feel right).

Merged.

**Pull request merged**